sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/TypeUtils.scala

```diff
@@ -32,6 +32,14 @@ object TypeUtils {
     }
   }
 
+  def checkForNumericAndIntervalExpr(t: DataType, caller: String): TypeCheckResult = {
```

Comment:
Now #7348  is in, just create a TypeCollection that contains all the numeric types as well as interval type.